src/realm/object-store/c_api/error.cpp

```diff
@@ -70,7 +76,7 @@ bool ErrorStorage::operator==(const ErrorStorage& other) const noexcept
     return m_err->error == other.m_err->error && m_message_buf == other.m_message_buf;
 }
 
-void ErrorStorage::assign(std::exception_ptr eptr) noexcept
+void ErrorStorage::assign(std::exception_ptr eptr, void* usercode_error) noexcept
```

Comment:
No need to pass usercode_error as parameter. It should be available in the CallbackFailed object if this is what is the current error.